From ae2ab30644815abefaa22f318ccbc9bd4e97c5b0 Mon Sep 17 00:00:00 2001
From: Tri Luong <tri.luong.te@rvc.renesas.com>
Date: Sat, 11 Aug 2018 00:26:09 +0700
Subject: [PATCH 614/642] pfc-r8a77470: Fix wrong definition I2C4, AVB pin mux.

Add missing definition USB0_PWEN0 pin mux and AVB_COL pin function.

Remove wrong definition TANS1, TANS2.

Fix wrong name MSIOF2_SS1, MSIOF2_SS2, SCIF3_SCK pin function.

Signed-off-by: Tri Luong <tri.luong.te@rvc.renesas.com>
---
 drivers/pinctrl/sh-pfc/pfc-r8a77470.c | 70 +++++++++++++++++++----------------
 1 file changed, 39 insertions(+), 31 deletions(-)

diff --git a/drivers/pinctrl/sh-pfc/pfc-r8a77470.c b/drivers/pinctrl/sh-pfc/pfc-r8a77470.c
index 24db3a6..bbbd6b2 100644
--- a/drivers/pinctrl/sh-pfc/pfc-r8a77470.c
+++ b/drivers/pinctrl/sh-pfc/pfc-r8a77470.c
@@ -109,8 +109,10 @@ enum {
 	FN_D13,	FN_MSIOF2_SYNC_A, FN_RX4_C,
 
 	/* IPSR3 */
-	FN_D14, FN_MSIOF2_SS1_A, FN_TX4_C, FN_CAN1_RX_B, FN_AVB_AVTP_CAPTURE_A,
-	FN_D15, FN_MSIOF2_SS2_A, FN_PWM4_A, FN_CAN1_TX_B, FN_IRQ2, FN_AVB_AVTP_MATCH_A,
+	FN_D14, FN_MSIOF2_SS1, FN_TX4_C, FN_CAN1_RX_B,
+	FN_AVB_AVTP_CAPTURE_A,
+	FN_D15, FN_MSIOF2_SS2, FN_PWM4_A, FN_CAN1_TX_B,
+	FN_IRQ2, FN_AVB_AVTP_MATCH_A,
 	FN_QSPI0_SPCLK, FN_WE0_N,
 	FN_QSPI0_MOSI_IO0, FN_BS_N,
 	FN_QSPI0_MISO_IO1, FN_RD_WR_N,
@@ -211,7 +213,7 @@ enum {
 	/* IPSR13 */
 	FN_SD2_DATA3, FN_TX2_A, FN_DU1_DB1, FN_SSI_WS9_B,
 	FN_SD2_CD, FN_SCIF2_SCK_A, FN_DU1_DB2, FN_SSI_SCK9_B,
-	FN_SD2_WP, FN_SCIF3_SCK_A, FN_DU1_DB3, FN_SSI_SDATA9_B,
+	FN_SD2_WP, FN_SCIF3_SCK, FN_DU1_DB3, FN_SSI_SDATA9_B,
 	FN_RX3_A, FN_SCL1_C, FN_MSIOF1_RXD_B, FN_DU1_DB4, FN_AUDIO_CLKA_C, FN_SSI_SDATA4_B,
 	FN_TX3_A, FN_SDA1_C, FN_MSIOF1_TXD_B, FN_DU1_DB5, FN_AUDIO_CLKB_C, FN_SSI_WS4_B,
 	FN_SCL2_A, FN_MSIOF1_SCK_B, FN_DU1_DB6, FN_AUDIO_CLKC_C, FN_SSI_SCK4_B,
@@ -252,8 +254,8 @@ enum {
 	FN_SSI_SCK9_A, FN_RX2_B, FN_SCL3_E, FN_EX_WAIT1, FN_VI0_DATA5_VI0_B5,
 	FN_SSI_WS9_A, FN_TX2_B, FN_SDA3_E, FN_VI0_DATA6_VI0_B6,
 	FN_SSI_SDATA9_A, FN_SCIF2_SCK_B, FN_PWM2_D, FN_VI0_DATA7_VI0_B7,
-	FN_AUDIO_CLKA_A, FN_SCL0_B, FN_TANS1, FN_VI0_CLKENB,
-	FN_AUDIO_CLKB_A, FN_SDA0_B, FN_TANS2, FN_VI0_FIELD,
+	FN_AUDIO_CLKA_A, FN_SCL0_B, FN_VI0_CLKENB,
+	FN_AUDIO_CLKB_A, FN_SDA0_B, FN_VI0_FIELD,
 	FN_AUDIO_CLKC_A, FN_SCL4_B, FN_VI0_HSYNC_N,
 	FN_AUDIO_CLKOUT_A, FN_SDA4_B, FN_VI0_VSYNC_N,
 
@@ -323,9 +325,11 @@ enum {
 	/* IPSR1 */
 	MMC0_D4_MARK, SD1_CD_MARK,
 	MMC0_D5_MARK, SD1_WP_MARK,
-	D0_MARK, SCL3_B_MARK, RX5_B_MARK, IRQ4_MARK, MSIOF2_RXD_C_MARK,	SSI_SDATA5_B_MARK,
+	D0_MARK, SCL3_B_MARK, RX5_B_MARK, IRQ4_MARK,
+	MSIOF2_RXD_C_MARK, SSI_SDATA5_B_MARK,
 	D1_MARK, SDA3_B_MARK, TX5_B_MARK, MSIOF2_TXD_C_MARK, SSI_WS5_B_MARK,
-	D2_MARK, RX4_B_MARK, SCL0_D_MARK, PWM1_C_MARK, MSIOF2_SCK_C_MARK, SSI_SCK5_B_MARK,
+	D2_MARK, RX4_B_MARK, SCL0_D_MARK, PWM1_C_MARK, MSIOF2_SCK_C_MARK,
+	SSI_SCK5_B_MARK,
 	D3_MARK, TX4_B_MARK, SDA0_D_MARK, PWM0_A_MARK, MSIOF2_SYNC_C_MARK,
 	D4_MARK, IRQ3_MARK, TCLK1_A_MARK, PWM6_C_MARK,
 	D5_MARK, HRX2_MARK, SCL1_B_MARK, PWM2_C_MARK, TCLK2_B_MARK,
@@ -341,8 +345,10 @@ enum {
 	D13_MARK, MSIOF2_SYNC_A_MARK, RX4_C_MARK,
 
 	/* IPSR3 */
-	D14_MARK, MSIOF2_SS1_A_MARK, TX4_C_MARK, CAN1_RX_B_MARK, AVB_AVTP_CAPTURE_A_MARK,
-	D15_MARK, MSIOF2_SS2_A_MARK, PWM4_A_MARK, CAN1_TX_B_MARK, IRQ2_MARK, AVB_AVTP_MATCH_A_MARK,
+	D14_MARK, MSIOF2_SS1_MARK, TX4_C_MARK, CAN1_RX_B_MARK,
+	AVB_AVTP_CAPTURE_A_MARK,
+	D15_MARK, MSIOF2_SS2_MARK, PWM4_A_MARK, CAN1_TX_B_MARK,
+	IRQ2_MARK, AVB_AVTP_MATCH_A_MARK,
 	QSPI0_SPCLK_MARK, WE0_N_MARK,
 	QSPI0_MOSI_IO0_MARK, BS_N_MARK,
 	QSPI0_MISO_IO1_MARK, RD_WR_N_MARK,
@@ -443,7 +449,7 @@ enum {
 	/* IPSR13 */
 	SD2_DATA3_MARK, TX2_A_MARK, DU1_DB1_MARK, SSI_WS9_B_MARK,
 	SD2_CD_MARK, SCIF2_SCK_A_MARK, DU1_DB2_MARK, SSI_SCK9_B_MARK,
-	SD2_WP_MARK, SCIF3_SCK_A_MARK, DU1_DB3_MARK, SSI_SDATA9_B_MARK,
+	SD2_WP_MARK, SCIF3_SCK_MARK, DU1_DB3_MARK, SSI_SDATA9_B_MARK,
 	RX3_A_MARK, SCL1_C_MARK, MSIOF1_RXD_B_MARK, DU1_DB4_MARK, AUDIO_CLKA_C_MARK, SSI_SDATA4_B_MARK,
 	TX3_A_MARK, SDA1_C_MARK, MSIOF1_TXD_B_MARK, DU1_DB5_MARK, AUDIO_CLKB_C_MARK, SSI_WS4_B_MARK,
 	SCL2_A_MARK, MSIOF1_SCK_B_MARK, DU1_DB6_MARK, AUDIO_CLKC_C_MARK, SSI_SCK4_B_MARK,
@@ -484,8 +490,8 @@ enum {
 	SSI_SCK9_A_MARK, RX2_B_MARK, SCL3_E_MARK, EX_WAIT1_MARK, VI0_DATA5_VI0_B5_MARK,
 	SSI_WS9_A_MARK, TX2_B_MARK, SDA3_E_MARK, VI0_DATA6_VI0_B6_MARK,
 	SSI_SDATA9_A_MARK, SCIF2_SCK_B_MARK, PWM2_D_MARK, VI0_DATA7_VI0_B7_MARK,
-	AUDIO_CLKA_A_MARK, SCL0_B_MARK, TANS1_MARK, VI0_CLKENB_MARK,
-	AUDIO_CLKB_A_MARK, SDA0_B_MARK,	TANS2_MARK, VI0_FIELD_MARK,
+	AUDIO_CLKA_A_MARK, SCL0_B_MARK, VI0_CLKENB_MARK,
+	AUDIO_CLKB_A_MARK, SDA0_B_MARK,	VI0_FIELD_MARK,
 	AUDIO_CLKC_A_MARK, SCL4_B_MARK, VI0_HSYNC_N_MARK,
 	AUDIO_CLKOUT_A_MARK, SDA4_B_MARK, VI0_VSYNC_N_MARK,
 
@@ -606,12 +612,12 @@ static const u16 pinmux_data[] = {
 
 	/* IPSR3 */
 	PINMUX_IPSR_DATA(IP3_3_0, D14),
-	PINMUX_IPSR_DATA(IP3_3_0, MSIOF2_SS1_A),
+	PINMUX_IPSR_DATA(IP3_3_0, MSIOF2_SS1),
 	PINMUX_IPSR_MSEL(IP3_3_0, TX4_C, SEL_SCIF4_2),
 	PINMUX_IPSR_MSEL(IP3_3_0, CAN1_RX_B, SEL_CAN1_1),
 	PINMUX_IPSR_MSEL(IP3_3_0, AVB_AVTP_CAPTURE_A, SEL_AVB_0),
 	PINMUX_IPSR_DATA(IP3_7_4, D15),
-	PINMUX_IPSR_DATA(IP3_7_4, MSIOF2_SS2_A),
+	PINMUX_IPSR_DATA(IP3_7_4, MSIOF2_SS2),
 	PINMUX_IPSR_DATA(IP3_7_4, PWM4_A),
 	PINMUX_IPSR_MSEL(IP3_7_4, CAN1_TX_B, SEL_CAN1_1),
 	PINMUX_IPSR_DATA(IP3_7_4, IRQ2),
@@ -935,7 +941,7 @@ static const u16 pinmux_data[] = {
 	PINMUX_IPSR_DATA(IP13_7_4, DU1_DB2),
 	PINMUX_IPSR_MSEL(IP13_7_4, SSI_SCK9_B, SEL_SSI9_1),
 	PINMUX_IPSR_DATA(IP13_11_8, SD2_WP),
-	PINMUX_IPSR_DATA(IP13_11_8, SCIF3_SCK_A),
+	PINMUX_IPSR_DATA(IP13_11_8, SCIF3_SCK),
 	PINMUX_IPSR_DATA(IP13_11_8, DU1_DB3),
 	PINMUX_IPSR_MSEL(IP13_11_8, SSI_SDATA9_B, SEL_SSI9_1),
 	PINMUX_IPSR_MSEL(IP13_15_12, RX3_A, SEL_SCIF3_0),
@@ -1079,11 +1085,9 @@ static const u16 pinmux_data[] = {
 	PINMUX_IPSR_DATA(IP17_11_8, VI0_DATA7_VI0_B7),
 	PINMUX_IPSR_MSEL(IP17_15_12, AUDIO_CLKA_A, SEL_ADGA_0),
 	PINMUX_IPSR_MSEL(IP17_15_12, SCL0_B, SEL_I2C00_1),
-	PINMUX_IPSR_DATA(IP17_15_12, TANS1),
 	PINMUX_IPSR_DATA(IP17_15_12, VI0_CLKENB),
 	PINMUX_IPSR_MSEL(IP17_19_16, AUDIO_CLKB_A, SEL_ADGB_0),
 	PINMUX_IPSR_MSEL(IP17_19_16, SDA0_B, SEL_I2C00_1),
-	PINMUX_IPSR_DATA(IP17_19_16, TANS2),
 	PINMUX_IPSR_DATA(IP17_19_16, VI0_FIELD),
 	PINMUX_IPSR_MSEL(IP17_23_20, AUDIO_CLKC_A, SEL_ADGC_0),
 	PINMUX_IPSR_MSEL(IP17_23_20, SCL4_B, SEL_I2C04_1),
@@ -1155,19 +1159,20 @@ static const unsigned int avb_mdio_mux[] = {
 };
 static const unsigned int avb_gmii_pins[] = {
 	/* RXD 0-7 */
-	RCAR_GP_PIN(3, 14), RCAR_GP_PIN(3, 15), RCAR_GP_PIN(3, 16),
-	RCAR_GP_PIN(3, 27), RCAR_GP_PIN(3, 28), RCAR_GP_PIN(3, 29),
-	RCAR_GP_PIN(4, 0),  RCAR_GP_PIN(5, 22),
+	RCAR_GP_PIN(3, 2), RCAR_GP_PIN(3, 3), RCAR_GP_PIN(3, 4),
+	RCAR_GP_PIN(3, 5), RCAR_GP_PIN(3, 6), RCAR_GP_PIN(3, 7),
+	RCAR_GP_PIN(3, 8), RCAR_GP_PIN(3, 9),
 	/* RX */
 	RCAR_GP_PIN(3, 10), RCAR_GP_PIN(3, 0), RCAR_GP_PIN(3, 1),
 
 	/* TXD 0-7 */
-	RCAR_GP_PIN(3, 2), RCAR_GP_PIN(3, 3), RCAR_GP_PIN(3, 4),
-	RCAR_GP_PIN(3, 5), RCAR_GP_PIN(3, 6), RCAR_GP_PIN(3, 7),
-	RCAR_GP_PIN(3, 8), RCAR_GP_PIN(3, 9),
+	RCAR_GP_PIN(3, 14), RCAR_GP_PIN(3, 15), RCAR_GP_PIN(3, 16),
+	RCAR_GP_PIN(3, 27), RCAR_GP_PIN(3, 28), RCAR_GP_PIN(3, 29),
+	RCAR_GP_PIN(4, 0),  RCAR_GP_PIN(5, 22),
 
 	RCAR_GP_PIN(5, 17),  RCAR_GP_PIN(4, 1),  RCAR_GP_PIN(3, 11),
 	RCAR_GP_PIN(3, 13), RCAR_GP_PIN(5, 23), RCAR_GP_PIN(3, 12),
+	RCAR_GP_PIN(5, 18),
 };
 static const unsigned int avb_gmii_mux[] = {
 	AVB_RXD0_MARK, AVB_RXD1_MARK, AVB_RXD2_MARK,
@@ -1181,6 +1186,7 @@ static const unsigned int avb_gmii_mux[] = {
 
 	AVB_CRS_MARK, AVB_GTX_CLK_MARK, AVB_GTXREFCLK_MARK,
 	AVB_TX_EN_MARK, AVB_TX_ER_MARK, AVB_TX_CLK_MARK,
+	AVB_COL_MARK,
 };
 /* - HSCIF1 ----------------------------------------------------------------- */
 static const unsigned int hscif1_data_pins[] = {
@@ -1246,7 +1252,7 @@ static const unsigned int i2c3_c_mux[] = {
 /* - I2C4 ------------------------------------------------------------------- */
 static const unsigned int i2c4_e_pins[] = {
 	/* SCL, SDA */
-	RCAR_GP_PIN(5, 6), RCAR_GP_PIN(5, 7),
+	RCAR_GP_PIN(5, 7), RCAR_GP_PIN(5, 6),
 };
 static const unsigned int i2c4_e_mux[] = {
 	SCL4_E_MARK, SDA4_E_MARK,
@@ -1308,7 +1314,7 @@ static const unsigned int msiof2_ss1_a_pins[] = {
 	RCAR_GP_PIN(1, 14),
 };
 static const unsigned int msiof2_ss1_a_mux[] = {
-	MSIOF2_SS1_A_MARK,
+	MSIOF2_SS1_MARK,
 };
 static const unsigned int msiof2_rx_a_pins[] = {
 	/* RXD */
@@ -1447,9 +1453,11 @@ static const unsigned int ssi9_b_data_mux[] = {
 };
 /* - USB0 ------------------------------------------------------------------- */
 static const unsigned int usb0_pins[] = {
+	RCAR_GP_PIN(0, 0), /* PWEN */
 	RCAR_GP_PIN(0, 1), /* OVC */
 };
 static const unsigned int usb0_mux[] = {
+	USB0_PWEN_MARK,
 	USB0_OVC_MARK,
 };
 /* - USB1 ------------------------------------------------------------------- */
@@ -2054,10 +2062,10 @@ static const struct pinmux_cfg_reg pinmux_config_regs[] = {
 		FN_QSPI0_SPCLK, FN_WE0_N, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
 		0, 0,
 		/* IP3_7_4 [4] */
-		FN_D15, FN_MSIOF2_SS2_A, FN_PWM4_A, 0, FN_CAN1_TX_B, FN_IRQ2,
+		FN_D15, FN_MSIOF2_SS2, FN_PWM4_A, 0, FN_CAN1_TX_B, FN_IRQ2,
 		FN_AVB_AVTP_MATCH_A, 0, 0, 0, 0, 0, 0, 0, 0, 0,
 		/* IP3_3_0 [4] */
-		FN_D14, FN_MSIOF2_SS1_A, 0, FN_TX4_C, FN_CAN1_RX_B,
+		FN_D14, FN_MSIOF2_SS1, 0, FN_TX4_C, FN_CAN1_RX_B,
 		0, FN_AVB_AVTP_CAPTURE_A,
 		0, 0, 0, 0, 0, 0, 0, 0, 0, }
 	},
@@ -2328,7 +2336,7 @@ static const struct pinmux_cfg_reg pinmux_config_regs[] = {
 		FN_AUDIO_CLKA_C, FN_SSI_SDATA4_B, 0, 0, 0, 0, 0, 0, 0, 0,
 		0, 0,
 		/* IP13_11_8 [4] */
-		FN_SD2_WP, FN_SCIF3_SCK_A, 0, FN_DU1_DB3, FN_SSI_SDATA9_B, 0,
+		FN_SD2_WP, FN_SCIF3_SCK, 0, FN_DU1_DB3, FN_SSI_SDATA9_B, 0,
 		0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
 		/* IP13_7_4 [4] */
 		FN_SD2_CD, FN_SCIF2_SCK_A, 0, FN_DU1_DB2, FN_SSI_SCK9_B, 0, 0,
@@ -2430,10 +2438,10 @@ static const struct pinmux_cfg_reg pinmux_config_regs[] = {
 		FN_AUDIO_CLKC_A, FN_SCL4_B, 0, 0, 0, 0,
 		FN_VI0_HSYNC_N, 0, 0, 0, 0, 0, 0, 0, 0, 0,
 		/* IP17_19_16 [4] */
-		FN_AUDIO_CLKB_A, FN_SDA0_B, 0, FN_TANS2, 0, 0,
+		FN_AUDIO_CLKB_A, FN_SDA0_B, 0, 0, 0, 0,
 		FN_VI0_FIELD, 0, 0, 0, 0, 0, 0, 0, 0, 0,
 		/* IP17_15_12 [4] */
-		FN_AUDIO_CLKA_A, FN_SCL0_B, 0, FN_TANS1, 0, 0,
+		FN_AUDIO_CLKA_A, FN_SCL0_B, 0, 0, 0, 0,
 		FN_VI0_CLKENB, 0, 0, 0, 0, 0, 0, 0, 0, 0,
 		/* IP17_11_8 [4] */
 		FN_SSI_SDATA9_A, FN_SCIF2_SCK_B, FN_PWM2_D, 0, 0, 0,
@@ -2606,4 +2614,4 @@ const struct sh_pfc_soc_info r8a77470_pinmux_info = {
 	.pinmux_data = pinmux_data,
 	.pinmux_data_size = ARRAY_SIZE(pinmux_data),
 };
-#endif
\ No newline at end of file
+#endif
-- 
2.7.4

